csrc/scheduler/reduction_utils.cpp

```diff
@@ -736,18 +736,38 @@ class PersistentBufferProjector {
             (int)buffer_i, fusion_->inputs());
       }
     } else {
+      // visit consumer before producer. e.g.
```

Comment:
This `project` function is getting bigger. Can you do some cleanup? Just separating it to `projectToInputs()` and `projectToProducers` would be enough.

Reply:
changed.